app/src/main/java/com/android/sample/ui/listActivities/LikedActivities.kt

```diff
@@ -143,7 +143,11 @@ fun ActivityCard2(
     allActivities: List<Activity>,
 ) {
 
-  val activity = allActivities.filter { act -> act.uid == activityId }[0]
+  val activity = allActivities.find { act -> act.uid == activityId }
```

Comment:
Nice catch for using find!

You can further improve conciseness by handling the null case with the "?:" as follows:
 allActivities.find { it.uid == activityId } ?: run {...}